orion/pipelines/azure.json

```diff
@@ -0,0 +1,42 @@
+{
+    "primitives": [
+        "mlprimitives.custom.timeseries_preprocessing.time_segments_aggregate",
+        "sklearn.impute.SimpleImputer",
+        "sklearn.preprocessing.MinMaxScaler",
+        "orion.primitives.azure_anomaly_detector.split_sequence",
+        "orion.primitives.azure_anomaly_detector.detect_anomalies"
+    ],
+    "init_params": {
+        "mlprimitives.custom.timeseries_preprocessing.time_segments_aggregate#1": {
+            "time_column": "timestamp",
+            "interval": 21600,
+            "method": "mean"
+        },
+        "sklearn.preprocessing.MinMaxScaler#1": {
+            "feature_range": [
+              -1,
+              1
+            ]
+        },
+        "orion.primitives.azure_anomaly_detector.split_sequence#1": {
+            "target_column": 0,
+            "sequence_size": 6000,
+            "pad_size": 2640
+        },
+        "orion.primitives.azure_anomaly_detector.detect_anomalies#1": {
+            "subscription_key": "",
+            "endpoint": "https://ad2.cognitiveservices.azure.com/",
```

Comment:
Maybe this could be set as default in the primitive itself, so new pipelines that use it do not need to set the endpoint again.

Also, regarding subscription_key, it would interesting if we can get that as input `init_params` instead of editing the JSON every time. It would imply adding the ability to pass params in other places, like `benchmark` and `analysis` modules, but it would be cleaner and also avoid the risk to accidentally commit a private key (this could be done in a later PR, just dropping the comment here)

Reply:
good point, actually this should be removed permanently since the endpoints should be different for each user, my mistake. 

There is one thing that's particularly interesting for a pipeline in this case. I need to keep a place holder for `subscription_key` and `endpoint` within the `pipeline.json` such that mlpipeline loads correctly (`self.blocks = self._build_blocks()` fails if a parameter without a default value is not instantiated), then I override the `init_params` in the json with hyperparameters passed to the `analyze` or `benchmark` function. I will upload a tutorial alongside this PR to showcase how this is achieved.

Reply:
> There is one thing that's particularly interesting for a pipeline in this case. I need to keep a place holder for `subscription_key` and `endpoint` within the `pipeline.json` such that mlpipeline loads correctly (`self.blocks = self._build_blocks()` fails if a parameter without a default value is not instantiated), then I override the `init_params` in the json with hyperparameters passed to the `analyze` or `benchmark` function. I will upload a tutorial alongside this PR to showcase how this is achieved.

Interesting. Supposedly, these type of fixed parameters without defaults should be able to be passed as `init_params` when the pipeline is loaded, but the way the `analyze` and `benchmark` functions are set, there are no `init_params` to pass and all you can do is update `hyperparams` _after_ the pipeline has already been created.

I think we can go ahead as it is, but it would be interesting to keep this in mind and add `init_params` to these (or other) functions in the future, so one can pass these or any other things to the pipelines.

Also, btw, notice that `MLPipeline.load` was deprecated in the latest MLBlocks release and that one can do `MLPipeline(pipeline_spec, init_params=...)` in all cases, whether `pipeline_spec` is a pipeline `dict`, a pipeline name, a path to a JSON or a list of primitives.

Reply:
Great, I will create an issue for this.